homeassistant/components/climate/intesishome.py

```diff
@@ -0,0 +1,350 @@
+STATE_DRY = 'Dry'
+STATE_AUTO = 'Auto'
+STATE_QUIET = 'Quiet'
+STATE_LOW = 'Low'
+STATE_MEDIUM = 'Medium'
+STATE_HIGH = 'High'
+STATE_OFF = 'Off'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_STRUCTURE): vol.All(cv.ensure_list, cv.string),
+})
+
+# Return cached results if last scan time was less than this value.
+# If a persistent connection is established for the controller,
+# changes to values are in realtime.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+
+try:
+    # pylint: disable=unused-import
+    from asyncio import ensure_future
+except ImportError:
+    # Python 3.4.3 and ealier has this as async
+    # pylint: disable=unused-import
+    from asyncio import async
+    ensure_future = async
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IntesisHome interface."""
+    from pyintesishome import IntesisHome
+
+    _user = config.get(CONF_USERNAME)
+    _pass = config.get(CONF_PASSWORD)
+
+    if 'climate_intesishome' not in hass.data:
+        hass.data['climate_intesishome'] = IntesisHome(_user, _pass, hass.loop)
+
+    intesishome = hass.data['climate_intesishome']
+    intesishome.connect()
+
+    if intesishome.error_message:
+        persistent_notification.create(hass, intesishome.error_message,
+                                       "IntesisHome Error", 'intesishome')
+    add_devices([IntesisAC(deviceid, device, hass)
+                 for deviceid, device in intesishome.get_devices().items()])
+    return True
+
+
+class IntesisAC(ClimateDevice):
+    """Representation of an IntesisHome thermostat."""
+
+    def __init__(self, deviceid, device, hass):
+        """Initialize the thermostat."""
+        _LOGGER.debug('Added climate device with state: %s', repr(device))
+        self._intesishome = hass.data.get('climate_intesishome')
+
+        self._deviceid = deviceid
+        self._devicename = device['name']
+
+        self._max_temp = None
+        self._min_temp = None
+        self._target_temp = None
+        self._current_temp = None
+        self._run_hours = None
+        self._rssi = None
+        self._swing = None
+        self._has_swing_control = False
+
+        self._power = STATE_UNKNOWN
+        self._fan_speed = STATE_UNKNOWN
+        self._current_operation = STATE_UNKNOWN
+
+        self._operation_list = [STATE_AUTO, STATE_COOL, STATE_HEAT, STATE_DRY,
+                                STATE_FAN, STATE_OFF]
+        self._fan_list = [STATE_AUTO, STATE_QUIET, STATE_LOW, STATE_MEDIUM,
+                          STATE_HIGH]
+        self._swing_list = ["Auto/Stop", "Swing", "Middle"]
+
+        # Best guess as which widget represents vertical swing control
+        if 42 in device.get('widgets'):
+            self._has_swing_control = True
+
+        self._intesishome.add_update_callback(self.update_callback)
+        self.update()
```

Comment:
Don't call `update` in init. Add a second argument `True` to the `add_devices` call in `setup_platform`. This will make sure `update` is called before adding the entity.